tck/features/expressions/graph/Graph8.feature

```diff
@@ -28,85 +28,125 @@
+      MATCH (n)
+      UNWIND keys(n) AS x
+      RETURN DISTINCT x AS theProps
       """
     Then the result should be, in any order:
-      | n               | b     |
-      | (:X {prop: 42}) | true  |
-      | (:X)            | false |
+      | theProps       |
+      | 'name'         |
+      | 'surname'      |
+      | 'otherName'    |
+      | 'otherSurname' |
     And no side effects
 
-  Scenario: [3] Property existence check on non-null node
+  Scenario: [3] Using `keys()` on a single node, empty result
     Given an empty graph
     And having executed:
       """
-      CREATE ({exists: 42})
+      CREATE ()
       """
     When executing query:
       """
       MATCH (n)
-      RETURN exists(n.missing),
-             exists(n.exists)
+      UNWIND keys(n) AS x
+      RETURN DISTINCT x AS theProps
       """
     Then the result should be, in any order:
-      | exists(n.missing) | exists(n.exists) |
-      | false             | true             |
+      | theProps |
     And no side effects
 
-  Scenario: [4] Property existence check on optional non-null node
+  Scenario: [4] Using `keys()` on an optionally matched node
     Given an empty graph
     And having executed:
       """
-      CREATE ({exists: 42})
+      CREATE ()
       """
     When executing query:
       """
       OPTIONAL MATCH (n)
-      RETURN exists(n.missing),
-             exists(n.exists)
+      UNWIND keys(n) AS x
+      RETURN DISTINCT x AS theProps
       """
     Then the result should be, in any order:
-      | exists(n.missing) | exists(n.exists) |
-      | false             | true             |
+      | theProps |
     And no side effects
 
-  Scenario: [5] Property existence check on null node
+  Scenario: [5] Using `keys()` on a relationship, non-empty result
     Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:KNOWS {status: 'bad', year: '2015'}]->()
+      """
     When executing query:
       """
-      OPTIONAL MATCH (n)
-      RETURN exists(n.missing)
+      MATCH ()-[r:KNOWS]-()
+      UNWIND keys(r) AS x
+      RETURN DISTINCT x AS theProps
+      """
+    Then the result should be, in any order:
+      | theProps |
+      | 'status' |
+      | 'year'   |
+    And no side effects
+
+  Scenario: [6] Using `keys()` on a relationship, empty result
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:KNOWS]->()
+      """
+    When executing query:
+      """
+      MATCH ()-[r:KNOWS]-()
+      UNWIND keys(r) AS x
+      RETURN DISTINCT x AS theProps
+      """
+    Then the result should be, in any order:
+      | theProps |
+    And no side effects
+
+  Scenario: [7] Using `keys()` on an optionally matched relationship
+    Given an empty graph
+    And having executed:
+      """
+      CREATE ()-[:KNOWS]->()
+      """
+    When executing query:
+      """
+      OPTIONAL MATCH ()-[r:KNOWS]-()
+      UNWIND keys(r) AS x
+      RETURN DISTINCT x AS theProps
```

Comment:
Those scenarios are handling edges while the feature is about node

Reply:
Good point. I have addressed this by renaming the feature accordingly to "Properties keys function". I have lines the respective feature names in map-related feature in the same commit.